src/mlpack/core/data/load_impl.hpp

```diff
@@ -403,36 +328,78 @@ bool Load(const std::string& filename,
   }
   else
   {
-    success = matrix.load(stream, ToArmaFileType(loadType));
+    success = matrix.load(stream, ToArmaFileType(opts.Format()));
   }
 
-  if (!success)
+  if (!opts.NoTranspose())
   {
-    Log::Info << std::endl;
-    Timer::Stop("loading_data");
-    if (fatal)
-      Log::Fatal << "Loading from '" << filename << "' failed." << std::endl;
-    else
-      Log::Warn << "Loading from '" << filename << "' failed." << std::endl;
+    // It seems that there is no direct way to use inplace_trans() on 
+    // sparse matrices. Therefore, we need the temporary SpMat
+    arma::SpMat<eT> tmp = matrix.t();
+    matrix = tmp;
```

Comment:
```suggestion
    // It seems that there is no direct way to use inplace_trans() on 
    // sparse matrices.
    matrix = matrix.t();
```

This will actually do exactly the same thing, it's just a little shorter.